egs2/TEMPLATE/tts1/tts.sh

```diff
@@ -295,22 +299,27 @@ if [ ${stage} -le 3 ] && [ ${stop_stage} -ge 3 ]; then
         # Remove short utterances
         _feats_type="$(<${data_feats}/${dset}/feats_type)"
         if [ "${_feats_type}" = raw ]; then
-            min_length=2560
+            # 256 is typical number of shift length
```

Comment:
Could you explain why 256?
16000 / 1000 = 16 samples / ms
10 ms * 16 = 160 samples?
Anyway, the number of shift sample length depends on the sampling rate and shift size (ms).

Reply:
256 is typical number for 16khz audio. 160, which is used for ASR recipe, is also okay.

More accurately, it's calculated from sampling rate and shift size, but it's impossible because shift size is not defined here since it's raw data.
We need two option, --max_wav_frames and --max_wav_seconds for each feature type,  but it looks verbose.
We don't need to calculate the accurate number here. It's enough to cut long/short data roughly.

Reply:
Sorry, I meant to say that is for ASR but I wrongly put the comment in the TTS script, but anyway both use the same 256, which is risky for me...

I don't recommend to make a rough estimate.
Note that we often use 8/16kHz data, and if you keep it 256, this will change the shift sample length two times different, and this threshold changes affect this pruning process very significantly...
Also, if such a number is embedded in `asr.sh` or `tts.sh`, people do not notice this issue, and they wrongly cut very large numbers of utterances unintentionally.

How about using `--max_wav_seconds` and `--min_wav_seconds`?
With the `fs` information, we can cut it with the exact number of samples.
In my experience, I actually set such thresholds based on the audio length (second), e.g, 20 seconds (which corresponds to 2,000 frames in the ASR case).
Why 20 seconds? This is a usual maximum length of the utterance (e.g., AMI and CHiME set this number during the manual transcription process).
Librispeedh does not follow this convention since it is a read speech, and I set 30 seconds by checking the histogram of the wave lengths of training files. (3,000 frames)

Reply:
I'll rethink

Reply:
I changed it to --min_wav_duration.
- If feats_type=raw: Calculate using fs
- If feats_type=fbank_pitch: Calculate using frame_shift. frame_shift is derived from fbank.conf if it can else using 10ms.